app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -20,22 +17,32 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     end
 
     api :GET, "/organizations/:organization_id/subscriptions", N_("List organization subscriptions")
-    api :GET, "/systems/:system_id/subscriptions", N_("List a system's subscriptions"), :deprecated => true
+    api :GET, "/systems/:system_id/subscriptions", N_("List a content host's subscriptions"), :deprecated => true
     api :GET, "/activation_keys/:activation_key_id/subscriptions", N_("List an activation key's subscriptions")
+    api :GET, "/subscriptions"
+    param_group :search, Api::V2::ApiController
     param :organization_id, :number, :desc => N_("Organization ID"), :required => true
-    param :system_id, String, :desc => N_("UUID of the system"), :required => false
+    param :system_id, String, :desc => N_("UUID of the content host"), :required => false
     param :activation_key_id, String, :desc => N_("Activation key ID"), :required => false
-    param_group :search, Api::V2::ApiController
+    param :available_for, String, :desc => N_("Object to show subscriptions available for, either 'content_host' or 'activation_key'"), :required => false
+    param :match_system, :bool, :desc => N_("Return subscriptions that match content_host")
+    param :match_installed, :bool, :desc => N_("Return subscriptions that match installed")
+    param :overlap, :bool, :desc => N_("Return subscriptions that don't overlap")
     def index
-      subscriptions = if @system
-                        index_system
-                      elsif @activation_key
-                        index_activation_key
-                      else
-                        index_organization
-                      end
+      respond(:collection => scoped_search(index_relation.uniq, :cp_id, :asc, :resource_class => Pool))
+    end
 
-      respond_for_index(:collection => subscriptions)
+    def index_relation
+      collection = Pool.readable
+      return available_for_system if params[:available_for] == "content_host"
+      return available_for_activation_key if params[:available_for] == "activation_key"
```

Comment:
If these return when called, the `collection = Pool.readable` could be moved below them as it wouldn't need to occur if these params were present.